src/tests/LitePsmJob-integration.t.sol

```diff
@@ -0,0 +1,166 @@
+
+    using GodMode for *;
+
+    uint256 constant MILLION_WAD = MILLION * WAD;
+
+    LitePsmLike public litePsm;
+    LitePsmJob public litePsmJob;
+    address public gem;
+    address public dai;
+    address public pocket;
+    address vat;
+    bytes32 ilk;
+
+    // --- Events ---
+    event Chug(uint256 wad);
+    event Fill(uint256 wad);
+    event Trim(uint256 wad);
+    event Work(bytes32 indexed network);
+
+    function setUpSub() virtual override internal {
+        litePsm = LitePsmLike(dss.chainlog.getAddress("MCD_LITE_PSM_USDC_A"));
+        pocket = dss.chainlog.getAddress("MCD_LITE_PSM_POCKET_USDC_A");
+        dai = dss.chainlog.getAddress("MCD_DAI");
+        litePsmJob = new LitePsmJob(address(sequencer), litePsm, MILLION_WAD, MILLION_WAD, MILLION_WAD);
+        gem = litePsm.gem();
+        ilk = litePsm.ilk();
+        vat = litePsm.vat();
+        // give auth access to this contract (caller) to vat for manipulating params
+        GodMode.setWard(vat, address(this), 1);
+    }
+
+    function test_fill() public {
+        // GodMode.setWard(address(litePsm), address(this), 1);
+        (uint256 Art,,, uint256 line,) = VatLike(vat).ilks(ilk);
+
+        // tArt must be greater than Art
+        // tArt = GemLike(gem).balanceOf(pocket) * GemConversionFactor + buf_;
+        uint256 GemConversionFactor = 10 ** (18 - GemLike(gem).decimals());
+        deal(gem, pocket, Art * 2 / GemConversionFactor);
+
+        // ilk line must be greater than Art
+        uint256 newLine = Art * 2 * RAY;
+        VatLike(vat).file(ilk, "line", newLine);
+        (Art,,, line,) = VatLike(vat).ilks(ilk);
+
+        // vat.Line() must be greater than vat.debt()
+        uint256 vatLine = VatLike(vat).Line();
+        uint256 vatDebt = VatLike(vat).debt();
+        VatLike(vat).file("Line", vatLine + vatDebt);
+
+        uint256 wad = litePsm.rush();
+        assertTrue(wad != 0);
+        (bool canWork, bytes memory args) = litePsmJob.workable(NET_A);
+        if (canWork){
+            (bytes4 fn) = abi.decode(args, (bytes4));
+            bytes4 encodedSelector = bytes4(abi.encode(litePsm.fill.selector));
+            assertEq(fn, encodedSelector);
+            vm.expectEmit(false, false, false, true);
+            emit Fill(wad);
+            vm.expectEmit(true, false, false, false);
+            emit Work(NET_A);
+            litePsmJob.work(NET_A, args);
+            wad = litePsm.rush();
+            assertEq(wad, 0);
+        }
+    }
+
+     function test_chug() public {
+        // the dai balance of LitePsm must be greater than the urn's art for this ilk
+        (, uint256 art) = VatLike(vat).urns(ilk, address(litePsm));
+        deal(dai, address(litePsm), art + 1); //must be greater than art so we dont have underflow
+        uint256 wad = litePsm.cut();
+        assertTrue(wad != 0);
+        (bool canWork, bytes memory args) = litePsmJob.workable(NET_A);
+        if (canWork){
```

Comment:
Same as above. This should be an assertion, not a conditional.

Reply:
fixed in [f05df30](https://github.com/makerdao/dss-cron/pull/25/commits/f05df30d9f7ea524df6d252cdcd68dc6767cead0)